### ISSUE TITLE
Some of mods in MODRINTH_PROJECTS isn't being downloaded

### ISSUE THREAD

**Issue by IntangibleW** [itzg/docker-minecraft-server#2227]
### Describe the problem

The mod "easyauth" and "fabrictailor" should be included, but mc-image-helper skipped downloading them.

### Container definition

```
version: "3"

services:
  Fabric:
    container_name: fabric
    image: itzg/minecraft-server:latest
    ports:
      - 25565:25565
      - 19132:19132/udp
    environment:
      TYPE: "FABRIC"
      MEMORY: "6G"
      USE_AIKAR_FLAGS: "TRUE"
      VERSION: "1.20.1"

      EULA: "TRUE"
      DIFFICULTY: "HARD"
      ENABLE_WHITELIST: "TRUE"
      ENFORCE_WHITELIST: "TRUE"
      MAX_PLAYERS: "20"
      FORCE_GAMEMODE: "TRUE"
      MAX_TICK_TIME: "-1"
      VIEW_DISTANCE: "16"
      ONLINE_MODE: "FALSE"
      ALLOW_FLIGHT: "FALSE"
      ENABLE_AUTOPAUSE: "TRUE"

      MODS: |
        https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/fabric
        https://ci.opencollab.dev/job/GeyserMC/job/Floodgate-Fabric/job/master/lastSuccessfulBuild/artifact/build/libs/floodgate-fabric.jar
      MODRINTH_PROJECTS: "fabric-api,lithium,easyauth,luckperms,fabrictailor"
      MODRINTH_DOWNLOAD_OPTIONAL_DEPENDENCIES: "TRUE"

    tty: true
    stdin_open: true
    restart: unless-stopped
    volumes:
      - ./Fabric:/data
```

### Container logs

```
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxrwx 1 1000 1000 4096 Jun 16 15:56 /data'
[init] Autopause functionality enabled
[init] Resolved version given 1.20.1 into 1.20.1 and major version 1.20
[init] Resolving type given FABRIC
[mc-image-helper] 16:01:34.351 INFO  : /data/mods/floodgate-fabric.jar is already up to date from https://ci.opencollab.dev/job/GeyserMC/job/Floodgate-Fabric/job/master/lastSuccessfulBuild/artifact/build/libs/floodgate-fabric.jar
[mc-image-helper] 16:01:34.584 INFO  : /data/mods/Geyser-Fabric.jar is already up to date from https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/fabric
[mc-image-helper] 16:01:36.883 INFO  : Downloading fabric-api-0.83.1+1.20.1.jar
[mc-image-helper] 16:01:37.726 INFO  : Downloading lithium-fabric-mc1.20.1-0.11.2.jar
[mc-image-helper] 16:01:39.785 INFO  : Downloading LuckPerms-Fabric-5.4.88.jar
[init] Enabling whitelist functionality
[init] Using Aikar's flags
[init] Setting initial memory to 6G and max to 6G
[init] Starting the Minecraft server...
Starting net.fabricmc.loader.impl.game.minecraft.BundlerClassPathCapture
[16:01:48] [main/INFO]: Loading Minecraft 1.20.1 with Fabric Loader 0.14.21
[16:01:49] [main/INFO]: Loading 61 mods:
- fabric-api 0.83.1+1.20.1
   |-- fabric-api-base 0.4.29+b04edc7a77
   |-- fabric-api-lookup-api-v1 1.6.34+4d8536c977
   |-- fabric-biome-api-v1 13.0.10+b3afc78b77
   |-- fabric-block-api-v1 1.0.9+e022e5d177
   |-- fabric-blockrenderlayer-v1 1.1.39+b3afc78b77
   |-- fabric-client-tags-api-v1 1.0.20+b3afc78b77
   |-- fabric-command-api-v1 1.2.32+f71b366f77
   |-- fabric-command-api-v2 2.2.11+b3afc78b77
   |-- fabric-commands-v0 0.2.49+df3654b377
   |-- fabric-containers-v0 0.1.61+df3654b377
   |-- fabric-content-registries-v0 4.0.7+b3afc78b77
   |-- fabric-convention-tags-v1 1.5.3+b3afc78b77
   |-- fabric-crash-report-info-v1 0.2.18+aeb40ebe77
   |-- fabric-data-generation-api-v1 12.1.11+b3afc78b77
   |-- fabric-dimensions-v1 2.1.51+b3afc78b77
   |-- fabric-entity-events-v1 1.5.21+b3afc78b77
   |-- fabric-events-interaction-v0 0.6.0+b3afc78b77
   |-- fabric-events-lifecycle-v0 0.2.61+df3654b377
   |-- fabric-game-rule-api-v1 1.0.38+b04edc7a77
   |-- fabric-item-api-v1 2.1.26+b3afc78b77
   |-- fabric-item-group-api-v1 4.0.7+b3afc78b77
   |-- fabric-key-binding-api-v1 1.0.36+fb8d95da77
   |-- fabric-keybindings-v0 0.2.34+df3654b377
   |-- fabric-lifecycle-events-v1 2.2.20+b3afc78b77
   |-- fabric-loot-api-v2 1.1.37+b3afc78b77
   |-- fabric-loot-tables-v1 1.1.41+9e7660c677
   |-- fabric-message-api-v1 5.1.6+b3afc78b77
   |-- fabric-mining-level-api-v1 2.1.47+b3afc78b77
   |-- fabric-models-v0 0.3.35+b3afc78b77
   |-- fabric-networking-api-v1 1.3.8+b3afc78b77
   |-- fabric-networking-v0 0.3.48+df3654b377
   |-- fabric-object-builder-api-v1 11.0.6+b3afc78b77
   |-- fabric-particles-v1 1.0.28+b3afc78b77
   |-- fabric-recipe-api-v1 1.0.18+b3afc78b77
   |-- fabric-registry-sync-v0 2.2.6+b3afc78b77
   |-- fabric-renderer-api-v1 3.0.1+b3afc78b77
   |-- fabric-renderer-indigo 1.3.1+b3afc78b77
   |-- fabric-renderer-registries-v1 3.2.44+df3654b377
   |-- fabric-rendering-data-attachment-v1 0.3.33+b3afc78b77
   |-- fabric-rendering-fluids-v1 3.0.26+b3afc78b77
   |-- fabric-rendering-v0 1.1.47+df3654b377
   |-- fabric-rendering-v1 3.0.6+b3afc78b77
   |-- fabric-resource-conditions-api-v1 2.3.5+ea08f9d877
   |-- fabric-resource-loader-v0 0.11.7+f7923f6d77
   |-- fabric-screen-api-v1 2.0.6+b3afc78b77
   |-- fabric-screen-handler-api-v1 1.3.27+b3afc78b77
   |-- fabric-sound-api-v1 1.0.12+b3afc78b77
   |-- fabric-transfer-api-v1 3.2.2+b3afc78b77
   \-- fabric-transitive-access-wideners-v1 4.2.0+b3afc78b77
- fabricloader 0.14.21
- floodgate 2.2.0-SNAPSHOT
   |-- adventure-platform-fabric 5.8.0-SNAPSHOT
   |    |-- net_kyori_adventure-api 4.13.0
   |    |-- net_kyori_adventure-key 4.13.0
   |    |-- net_kyori_adventure-platform-api 4.2.0
   |    |-- net_kyori_adventure-text-logger-slf4j 4.13.0
   |    |-- net_kyori_adventure-text-minimessage 4.13.0
   |    |-- net_kyori_adventure-text-serializer-gson 4.13.0
   |    |-- net_kyori_adventure-text-serializer-plain 4.13.0
   |    |-- net_kyori_examination-api 1.3.0
   |    \-- net_kyori_examination-string 1.3.0
   \-- cloud 1.8.3
        |-- cloud_commandframework_cloud-brigadier_ 1.8.3
        |-- cloud_commandframework_cloud-core_ 1.8.3
        |-- cloud_commandframework_cloud-services_ 1.8.3
        |-- fabric-permissions-api-v0 0.2-SNAPSHOT
        \-- io_leangen_geantyref_geantyref 1.3.13
- geyser-fabric 2.1.1-SNAPSHOT
   \-- fabric-permissions-api-v0 0.2-SNAPSHOT
- java 17
- lithium 0.11.2
- luckperms 5.4.88
   \-- fabric-permissions-api-v0 0.2-SNAPSHOT
- minecraft 1.20.1
[16:01:49] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/data/libraries/net/fabricmc/sponge-mixin/0.12.5+mixin.0.8.5/sponge-mixin-0.12.5+mixin.0.8.5.jar Service=Knot/Fabric Env=SERVER
[16:01:50] [main/INFO]: Compatibility level set to JAVA_17
[16:01:50] [main/INFO]: Loaded configuration file for Lithium: 115 options available, 0 override(s) found
[16:02:03] [main/INFO]: Loading extensions...
[16:02:03] [main/INFO]: Loaded 0 extension(s)
[16:02:04] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[16:02:09] [main/INFO]: Loaded 7 recipes
[16:02:09] [main/INFO]: Loaded 1271 advancements
[16:02:10] [main/INFO]: Applied 0 biome modifications to 0 of 64 new biomes in 2.212 ms
[16:02:10] [Server thread/INFO]:         __    
[16:02:10] [Server thread/INFO]:   |    |__)   LuckPerms v5.4.88
[16:02:10] [Server thread/INFO]:   |___ |      Running on Fabric - fabric@0.83.1+1.20.1
[16:02:10] [Server thread/INFO]: 
[16:02:10] [Server thread/INFO]: Loading configuration...
[16:02:10] [Server thread/INFO]: Loading storage provider... [H2]
[16:02:11] [Server thread/INFO]: Loading internal permission managers...
[16:02:11] [Server thread/INFO]: Performing initial data load...
[16:02:11] [Server thread/INFO]: Successfully enabled. (took 1318ms)
[16:02:11] [Server thread/INFO]: Starting minecraft server version 1.20.1
[16:02:11] [Server thread/INFO]: Loading properties
[16:02:11] [Server thread/INFO]: Default game type: SURVIVAL
[16:02:11] [Server thread/INFO]: Generating keypair
[16:02:11] [Server thread/INFO]: Starting Minecraft server on *:25565
[16:02:11] [Server thread/INFO]: Using epoll channel type
[16:02:12] [Server thread/WARN]: **** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
[16:02:12] [Server thread/WARN]: The server will make no attempt to authenticate usernames. Beware.
[16:02:12] [Server thread/WARN]: While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.
[16:02:12] [Server thread/WARN]: To change this, set "online-mode" to "true" in the server.properties file.
[16:02:12] [Server thread/INFO]: Preparing level "world"
[16:02:13] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[16:02:15] [Worker-Main-4/INFO]: Preparing spawn area: 0%
[16:02:15] [Worker-Main-4/INFO]: Preparing spawn area: 0%
[16:02:15] [Worker-Main-7/INFO]: Preparing spawn area: 0%
[16:02:15] [Worker-Main-5/INFO]: Preparing spawn area: 0%
[16:02:15] [Worker-Main-4/INFO]: Preparing spawn area: 0%
[16:02:16] [Worker-Main-7/INFO]: Preparing spawn area: 5%
[16:02:16] [Server thread/INFO]: Time elapsed: 2956 ms
[16:02:16] [Server thread/INFO]: Done (4.111s)! For help, type "help"
[16:02:16] [Server thread/INFO]: Starting remote control listener
[16:02:16] [Server thread/INFO]: Thread RCON Listener started
[16:02:16] [Server thread/INFO]: RCON running on 0.0.0.0:25575
[16:02:16] [Server thread/INFO]: Took 64ms to boot Floodgate
[16:02:16] [Server thread/INFO]: ******************************************
[16:02:16] [Server thread/INFO]: 
[16:02:16] [Server thread/INFO]: Loading Geyser version 2.1.1-SNAPSHOT (git-master-6591332)
[16:02:16] [Server thread/INFO]: 
[16:02:16] [Server thread/INFO]: ******************************************
[16:02:19] [Server thread/INFO]: Started Geyser on 0.0.0.0:19132
[16:02:20] [Server thread/INFO]: Done (3.076s)! Run /geyser help for help!
[16:02:20] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 3181ms or 63 ticks behind
[Autopause loop] MC Server listening for connections - pausing in 600 seconds

```

**Comment by itzg** [comment]
Thanks for reporting this. I'll look into it.

**Comment by itzg** [comment]
It turned out it is because those projects are not yet available for 1.20.1, just 1.20. (Screenshot below as of this time.) I am going to fix the download helper to detect that and fail with an explanation.



![Image](https://github.com/itzg/docker-minecraft-server/assets/988985/31c658b5-5a74-4383-8578-2db954a0b143)
